scripts/flashing/dlink_recovery_upload.py

```diff
@@ -0,0 +1,337 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: MIT
+#
+#   D-Link COVR-P2500 flash utility
+#
+#   Copyright (C) 2023 Daniel Linjama <daniel@dev.linjama.com>
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a copy of
+#   this software and associated documentation files (the “Software”), to deal in the
+#   Software without restriction, including without limitation the rights to use, copy,
+#   modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+#   and to permit persons to whom the Software is furnished to do so, subject to the
+#   following conditions:
+#
+#   The above copyright notice and this permission notice shall be included in all
+#   copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+#   INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+#   PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+#   HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+#   OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+#   SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+import argparse
+import ctypes
+import ctypes.wintypes
+import os
+import random
+import re
+import socket
+import struct
+import sys
+import time
+
+BUFFER_SIZE = 512
+ACK_TIMEOUT = 10
+
+
+if sys.platform in ("linux", "linux2"):
+    TCP_INFO_FMT = None
+    TCP_INFO_ITEMS = None
+
+    # Parse tcp_info structure from /usr/include/netinet/tcp.h
+    assert os.path.isfile("/usr/include/netinet/tcp.h"), "File not found: /usr/include/netinet/tcp.h"
+    with open("/usr/include/netinet/tcp.h", "r") as fh:
```

Comment:
This should be a stable user space ABI, I do not think you have to parse this header file.
Have you seen different versions of this structure?

Reply:
Changed to use `ctypes.Structure` implementation. [Musl](https://git.musl-libc.org/cgit/musl/tree/include/netinet/tcp.h#n195) have different tcp_info structure